api/openapi.yaml

```diff
@@ -1685,34 +1661,81 @@ components:
-              type: string
-              enum:
-                - Pending
-                - Discovered
-                - InProgress
-                - Aborted
-                - Failed
-                - Done
-            stateMessage:
-              description: Human-readable message indicating details about the last state transition.
-              type: string
-            stateReason:
-              description: Machine-readable, UpperCamelCase text indicating the reason for the condition's last transition.
-              type: string
-              enum:
-                - Aborted
-                - TimedOut
-                - OneOrMoreAssetFailedToScan
-                - DiscoveryFailed
-                - Unexpected
-                - NothingToScan
-                - Success
+            status:
+              $ref: '#/components/schemas/ScanStatus'
             summary:
               $ref: '#/components/schemas/ScanSummary'
         - $ref: '#/components/schemas/ScanTemplate'
 
+    ScanStatus:
+      type: object
+      required:
+        - state
+        - reason
+        - lastTransitionTime
+      properties:
+        state:
+          description: |
+            Describes the state of Scan.
+            
+            | State      | Description                                       |
+            | ---------- | ------------------------------------------------- |
+            | Pending    | Scan is waiting to be scheduled                   |
+            | Discovered | Assets have been found for the Scan               |
+            | InProgress | The AssetScans are running on the target machines |
+            | Aborted    | Scan has been aborted through the API             |
+            | Failed     | At least one AssetScan ended with an error        |
+            | Done       | Every AssetScan ended successfully                |
+          type: string
+          enum:
+            - Pending
+            - Discovered
+            - InProgress
+            - Aborted
+            - Failed
+            - Done
+        reason:
+          description: |
+            Machine readable reason for state transition.
+            
+            | State      | Reason            | Description                                                |
+            | ---------- | ----------------- | ---------------------------------------------------------- |
+            | Pending    | Created           | Scan is pending as it has been newly created               |
+            | Discovered | AssetsDiscovered  | Assets that fall into the scope of the Scan are discovered |
+            | InProgress | AssetScansRunning | AssetScans are currently running                           |
+            | Aborted    | Cancellation      | Scan has been aborted due to cancellation request          |
+            | Failed     | Cancellation      | Scan has been aborted due to cancellation request          |
```

Comment:
How do we get the scan being cancelled and ending up in `Aborted` or `Failed` state? I think it should be mentioned why it ends up in the given state due to cancellation. If there is no difference, then one can be removed (either _Aborted_ can act as _Failed_ state, or _Failed_ state should not include cancellation event/reason)

Reply:
The Scan first gets PATCH-ed to `Aborted` state and it starts updating the related AssetScans to `Aborted` state as well. After it finished updating every AssetScan it moves to `Failed` state. The difference before this PR was that the `Aborted` state of the Scan didn't had any Reason, and the Reason for the `Failed` state was `Aborted`, now both has `Cancellation` state. I extended the description with this information.